### ISSUE TITLE
Setting #none in the commit message will then cause all subsequent bumps to NOT bump the version

### ISSUE THREAD

**Issue by scott-doyland-burrows** [anothrNick/github-tag-action#194]
If I set ```#none``` in the commit message, then there is no new bump tag as expected.

However, all subsequent runs of the action then look at all the older commits and see ```#none``` (from the previous commit) and then no bump tag happens.

To start bump tagging again, I have to explicitly set a commit message such as ```#minor``` and all then starts to work as expected again - and the old ```#none``` is ignored for all future runs of the action, even when I do not supply an commit message that contains the bump type.

**Comment by sbe-arg** [comment]
@scott-doyland-burrows can you share some logs or screenshots? Ill also see if can replicate on my end

**Comment by sbe-arg** [comment]
ohhhh I see now.
By current implementation we read the full log of the branch if there are #major, #minor, #patch #none  

This becomes a "problem" if we intend to start a branch for a #minor bump and then the branch changes become a #major bump as the `case X in esac` does not follow order or commits when reading the log dump. https://github.com/anothrNick/github-tag-action/blob/master/entrypoint.sh#L123-L144

What could be done to help this situation is to have another flag to only read the last commit instead of the full log by default. altering this  https://github.com/anothrNick/github-tag-action/blob/master/entrypoint.sh#L100 @sammcj what do you think about this?

**Comment by sammcj** [comment]
That's a funny bug / oddity 😅.

@sbe-arg I think that's sensible, there might be other use cases for it too.

As long as we're not affecting the default behaviour I can't see it being a problem at least.

**Comment by scott-doyland-burrows** [comment]
As the GHA bump is reading the branch commits that it is bumping for, well, that branch is likely to be ```main``` so it will always be having a variation of ```#major```, ```#minor```, ```#patch```, ```#none```.  So I am not too clear what you mean by the example of:

*This becomes a "problem" if we intend to start a branch for a #minor bump and then the branch changes become a #major bump*

This is what happens:

I set the bump GHA to run when a push to ```main``` occurs,

1. I create a branch called ```branch1``` from ```main```
2. I commit with ```#none```
3. I PR into ```main```
4. There is no bump as expected
5. I create a branch called ```branch2``` from ```main```
6. I commit but do not specify any ```#xxxx``` at all
7. I PR in ```main```
8. The bump GHA finds the ```#none``` from the old commit into ```main``` and so does not bump

I think reading just the last commit sounds sensible, and if there is no ```#xxxx``` then just use the default bump.

However, the issue does only seem to occur with ```#none```.  If I follow the above steps but set ```#major``` rather than ```#none``` in step 2, then it reverts to the default bump once I reach step 8.

**Comment by sbe-arg** [comment]
Oh so you are not squashing the merges that will keep linear history. Okay Ill have a think as we need to refactor the documentation a bit as well. But we should implement the reading the last commit log flag.

**Comment by scott-doyland-burrows** [comment]
I am doing a ```squash and merge``` when I merge in the PR.

I created a new repo with a ```main``` branch.

1. I then setup a ```test01``` branch from ```main``` and squashed and merged with no ```#xxxx``` comments (single commit squashed and merged to ```main```)

2. Did the same for ```test02``` branch (single commit squashed and merged to ```main```)

3. Same for test for ```test03``` branch, but I added a comment of ```#none``` to the commit (single commit).

4. Then ```test04``` same as for 1 and 2 above, with no ```#xxx``` comment - and here is the screenshot below - see how it picked up on the ```#none``` from the commit for 3 above:

![image](https://user-images.githubusercontent.com/61512176/193752840-fb9a5e31-b2e4-48e8-9653-7fc9decc4300.png)
